mazegame/Assets/Scripts/MazeRunner/MazeRunner.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class MazeRunner : MonoBehaviour
+{
+
+    public float walkSpeed;
+    public float rotationSpeed;
+
+    public Transform rotationTransform;
+
+    Vector2 direction = Vector2.zero;
+
+    int targetX = 1;
+    int targetY = 1;
+
+    int currentX = 1;
+    int currentY = 1;
+
+    float currentAngle;
+    float lastAngle;
+
+    void Update()
+    {
+        bool targetReached = transform.position.x == targetX && transform.position.y == targetY;
+
+        currentX = Mathf.FloorToInt(transform.position.x);
+        currentY = Mathf.FloorToInt(transform.position.y);
+
+        direction.x = Input.GetAxisRaw("Horizontal");
+        direction.y = Input.GetAxisRaw("Vertical");
+
+        float angle = 0;
```

Comment:
Please use a more appropriate name for this. What does this angle belong to

Reply:
I haven't changed this.